### PR TITLE
Switching encryption library to c implementation

### PR DESCRIPTION
**Pull request by shuaiLiWang**
## What does this implement/fix? Explain your changes.
commit1：切换ecc库到 secp256k1的c实现
commit2：增加ios环境下加载库的路径
## Does this close any currently open issues?  
If it fixes a bug or resolves a feature request, be sure to link to that issue.
https://github.com/OneKeyHQ/TaskHub/issues/1419

## Pull request type
 
_Put an `x` in the boxes that apply_
- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe): 

## Where has this been tested?
Android

## Any other comments?
…

### CODE REVIEW

**Comment by taimanhui** [comment]
第一个commit是同步上游electrum的ecc.py么？我看整体很像不过没有一行行比对

**Comment by shuaiLiWang** [comment]
> 第一个commit是同步上游electrum的ecc.py么？我看整体很像不过没有一行行比对

是的

**Review by danielsocials — APPROVED** [review]

**Review by taimanhui — DISMISSED** [review]
我对比了上游的代码确认完全内容一致，对应当前上游的commit是
```
commit cad4e77853b1d3ae5fdaaa868c2fce67780a8870
Author: SomberNight <somber.night@protonmail.com>
Date:   Sat May 15 06:57:59 2021 +0200

    fix prev
```

```
$ git log upstream/master |head
commit cad4e77853b1d3ae5fdaaa868c2fce67780a8870
Author: SomberNight <somber.night@protonmail.com>
Date:   Sat May 15 06:57:59 2021 +0200

    fix prev

commit b2169b745e50612a95b70f630078c8fd6ec0be18
Author: SomberNight <somber.night@protonmail.com>
Date:   Sat May 15 06:31:03 2021 +0200

$ git diff upstream/master -- electrum/ecc_fast.py
$ git diff upstream/master -- electrum/ecc.py

```

File: contrib/deterministic-build/requirements-android.txt [hunk 1]
```diff
@@ -23,10 +23,11 @@ coincurve==13.0.0 \
     --hash=sha256:93e21f622e5db2cde54626f20f612b74a3bbe110137de7398d5e7cb3c7b84c5e \
     --hash=sha256:11d9f5c8e51766420b1d62e55705eeb70999d5f103ec0679205142f4f0ed4223 \
     --hash=sha256:bb15963112fbb35386dc1098c9fe39bfd9991e4d401b6bf9d68d0df7915a0b0f \
-    --hash=sha256:4c7617b18abbfdc1dcb396e552df23fbc79e90f5f741382c9a59a3ea039a612f 
+    --hash=sha256:4c7617b18abbfdc1dcb396e552df23fbc79e90f5f741382c9a59a3ea039a612f
 pycparser==2.19 \
     --hash=sha256:a988718abfad80b6b157acce7bf130a30876d27603738ac39f140993246b25b3
 
+PyNaCl==1.3.0 --hash=sha256:58bea623708dcc423e9203b33e125d9c8134709dd4ff8fb2747554935a9ccbed
```
Comment (huazhouwang) [inline contrib/deterministic-build/requirements-android.txt:30]: https://github.com/OneKeyHQ/electrum/pull/758 也添加了相同的库

**Comment by shuaiLiWang** [comment]
增加了第三个commit 修改了ios环境下加载库的路径

**Review by liyanhrxy — APPROVED** [review]

**Review by danielsocials — DISMISSED** [review]

**Review by taimanhui — APPROVED** [review]
这个还在测试吗？怎么还是draft的？

**Comment by shuaiLiWang** [comment]
> 这个还在测试吗？怎么还是draft的？

因为昨天晚上 android要发版 所以终止了合并 马上合并

**Review by liyanhrxy — APPROVED** [review]

**Review by huazhouwang — APPROVED** [review]

**Pull request merged**